src/hooks/useAutoClosableDropdown.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useRef } from 'react';
+import Dropdown, { DropdownItem } from '../components/Dropdown';
+
+const useAutoClosableDropdown = <T extends HTMLElement>(
+  items: Array<DropdownItem>,
+  elementRef: React.RefObject<T>
+) => {
+  const [shouldShow, setShouldShow] = React.useState(false);
+
+  // this is used as an event listener doesn't have access to the state
+  const sholdShowRef = useRef(shouldShow);
+
+  const updateVisibility = (value: boolean) => {
+    setShouldShow(value);
+    sholdShowRef.current = value;
+  };
+
+  const hideDropdownIfVisible = () => {
+    if (sholdShowRef.current === true) {
+      updateVisibility(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', hideDropdownIfVisible, true);
+
+    return () =>
+      document.removeEventListener('click', hideDropdownIfVisible, true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return {
+    renderDropdown: (
+      <Dropdown items={items} elementRef={elementRef} shouldShow={shouldShow} />
+    ),
+    visible: shouldShow,
+    showDropdown: updateVisibility,
+  };
+};
+
+export default useAutoClosableDropdown;
```

Comment:
Use a named export here, perhaps?

Reply:
Tried to, but ESLint won't stop complaining 😅 I could of course manually ignore it 🤔